### ISSUE TITLE
Errors in extensions are not returned to webmake

### ISSUE THREAD

**Issue by tswaters** [medikoo/modules-webmake#48]
I'm using webmake with (a customized) webmake-sass (to allow scss, see https://github.com/acdaniel/webmake-sass/pull/2). 

If there is an error in the compilation, the error is not returned to webmake and the css does not make it's way into the bundled JS file.

```sh
$ echo "\$bg:'#fff'; body {background:\$bad-bg;}" > style.scss
$ echo "require('./style')" > app.js
$ webmake --ext=sass app.js | grep 'style.scss' -A 4

                "style.scss": {
                        "value": {

                        }
                }
```

The following error should be returned: `source string:1: error: unbound variable $bad-bg`.  

webmake-sass (and webmake-less, for that matter) can reject the promise, but in cases where there is an error, it is not handled by webmake -- looking at the console, everything appears to be OK.

**Comment by medikoo** [comment]
Thanks for reporting it. It might be an issue with an extension itself. Anyway I'll look at it as soon as I'll get back from holiday (tomorrow)

**Comment by medikoo** [comment]
@tswaters I investigated it, and issue was caused by different versions of [deferred](https://www.npmjs.com/package/deferred) package as used by webmake, and webmake-sass.

Webmake uses v0.7 version of deferred, while plugin as not updated was still using v0.6.
Using promise objects from both versions interactively caused this nasty bug.

I've fixed deferred package, so it's no longer a case. So if you assure that you're Webmake implementation uses deferred at v0.7.2 you will see error thrown as expected.
